### PR TITLE
Quality: Parameterize SDK and test-data branches

### PR DESCRIPTION
**Pull request by typotter**
🎟️ Fixes FF-3089 towards FF-3085

👯‍♂️ **Related PRs**
- [sdk-test-data#57](https://github.com/Eppo-exp/sdk-test-data/pull/57)
- [php-sdk#38](https://github.com/Eppo-exp/php-sdk/pull/38)
- [dot-net-server-sdk#44](https://github.com/Eppo-exp/dot-net-server-sdk/pull/44)

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
_This change_
🚀 - Each SDK's testing workflow is enhanced into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), exposing parameters for SDK branch and the sdk-test-data branch to use in testing.
- The test workflow runs using the main branch of sdk-test-data on all main pushes and Pull Requests using the PR's branch

_External to this Change_
[sdk-test-data](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows) get two testing workflows.
1. ♻️  "Local Testing"- For all pull request changes, the "Local Testing" workflow calls the reusable SDK workflows, test results are recorded only in the sdk-test-data action. This is run using the main SDK branch and the "current" branch of workflow, i.e. the pull request branch. (SDK repo does not see/is not notified of test failures during PR lifecycle, only on push to main)
2. ♻️ 🧑‍💻 "Remote Testing" - On all pushes to the main branch of sdk-test-data, the SDK testing workflows are triggered to run within their respective repositories, alerting all subscribers of failed runs (repo is red/green).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ test-data:
 
 ## test
 .PHONY: test
-test: test
+test: test test-data
```
Comment (typotter) [inline Makefile:42]: unsure test data is always pulled before a test

File: .github/workflows/lint-test-sdk.yml [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
+        description: The branch of the SDK to test
+        required: false
+          
 jobs:
   lint-test-sdk:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          repository: Eppo-exp/node-server-sdk
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:29]: explicitly specify the SDK repository so when this workflow is reused, it's not trying to checkout the sdk-test-data repo here.

File: .github/workflows/lint-test-sdk.yml [hunk 3]
```diff
@@ -1,13 +1,33 @@
 name: Lint and Test SDK
+
+env:
+  SDK_BRANCH_NAME: ${{ inputs.sdk_branch  || github.head_ref || github.ref_name || 'main' }}
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:4]: allow `workflow_call` to spec an SDK branch OR use _head_ref_ or _ref_name_ which are github variables set to point to the local branch this workflow is running on (ex: a PR branch)

File: .github/workflows/lint-test-sdk.yml [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
+  TEST_DATA_BRANCH_NAME: ${{ inputs.test_data_branch || 'main' }}
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:5]: default to main for workflow triggers which do not take inputs.

File: .github/workflows/lint-test-sdk.yml [hunk 5]
```diff
@@ -19,21 +39,22 @@ jobs:
       - name: Install root dependencies
         run: yarn --frozen-lockfile
         working-directory: ./
-      - name: 'Set up GCP SDK for downloading test data'
-        uses: 'google-github-actions/setup-gcloud@v0'
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:23]: this action is deprecated; also gcloud is no longer needed for downloading test data

**Review by leoromanovsky — APPROVED** [review]

File: .github/workflows/lint-test-sdk.yml [hunk 6]
```diff
@@ -1,13 +1,33 @@
...
+env:
+  SDK_BRANCH_NAME: ${{ inputs.sdk_branch  || github.head_ref || github.ref_name || 'main' }}
+  TEST_DATA_BRANCH_NAME: ${{ inputs.test_data_branch || 'main' }}
+
 on:
   pull_request:
-    path: '**/*'
-
+    branches: [ "*" ]
+  workflow_dispatch:
```
Comment (aarsilv) [inline .github/workflows/lint-test-sdk.yml:10]: 👍

File: .github/workflows/lint-test-sdk.yml (hunk 5 above)
Comment (aarsilv) [reply .github/workflows/lint-test-sdk.yml:23]: 🪓

**Review by aarsilv — APPROVED** [review]
📈
